### ISSUE TITLE
Enhancement of logging output

### ISSUE THREAD

**Issue by HolQue** [test-fullautomation/python-pytestlog2db#21]
The feedback of an initial database access is:

```
Created test execution result for variant 'TT-PV-006' - version 'TT-VSW-006' successfully: ...
  Created test file result for classname '...' successfully: 15788
    Created test case result for test '...' successfully: 68927 (component: PyExColl)
```

The feedback of an additional database access in append mode is:

```
Append to existing test execution result UUID '...'.
  Created test file result for classname '...' successfully: 15796
    Created test case result for test '...' successfully: 69058 (component: GenPkgDoc)
```

In second case the variant and the version is missing, but would be very helpful also at end of console output.

Instead of:

`Append to existing test execution result UUID '...'.`

better:

`Append to existing test execution result for variant 'TT-PV-006' - version 'TT-VSW-006' - with UUID '...'.`

Belongs also to

https://github.com/test-fullautomation/robotframework-robotlog2db/issues/37

**Comment by HolQue** [comment]
Another small wish:

Please add a blank line in console before the final feedback. This would make the console output easier to read.

**Comment by ngoan1608** [comment]
> The feedback of an initial database access is:
> 
> ```
> Created test execution result for variant 'TT-PV-006' - version 'TT-VSW-006' successfully: ...
>   Created test file result for classname '...' successfully: 15788
>     Created test case result for test '...' successfully: 68927 (component: PyExColl)
> ```
> 
> The feedback of an additional database access in append mode is:
> 
> ```
> Append to existing test execution result UUID '...'.
>   Created test file result for classname '...' successfully: 15796
>     Created test case result for test '...' successfully: 69058 (component: GenPkgDoc)
> ```
> 
> In second case the variant and the version is missing, but would be very helpful also at end of console output.
> 
> Instead of:
> 
> `Append to existing test execution result UUID '...'.`
> 
> better:
> 
> `Append to existing test execution result for variant 'TT-PV-006' - version 'TT-VSW-006' - with UUID '...'.`
> 
> Belongs also to
> 
> [test-fullautomation/robotframework-robotlog2db#37](https://github.com/test-fullautomation/robotframework-robotlog2db/issues/37)

Hi Holger,

For append mode, the UUID is used to check for the existence of result in database.
The **variant** and **version_sw** information which are provided by command line, configuration *.json or test metadata will not be used and maybe different from the existing information in the database (has been created by previous import with that UUID).

I assume that want these information for browsing easily on WebApp but I think the UUID is enough.
In case you really want these information in console output, I can implement to get these information from database then use them for the console output instead of given information (command line arguments, config *.json, test metadata) when executing the import tool.

Thank you,
Ngoan

**Comment by test-fullautomation** [comment]
Hi Ngoan,
what happens if you use same UUID, but different version information in Append mode?
I would expect an error message?
Thank you,
 Thomas

**Comment by ngoan1608** [comment]
Hi Thomas,

With current implementation, only UUID is used to verify the existence of result in db and no error is raised now due to the difference between given _version_ ( or _project/variant_) and existing information in database when using append mode.

With your change request, there will be some changes in append mode behavior: besides checking the existence of UUID, then tool read db and compare with given information (_version_, _variant/project_, ... => need to be defined which information for comparison).


Thank you,
Ngoan

**Comment by HolQue** [comment]
I do not understand this. If `variant`and `version_sw` is provided to the database application also in append mode, why not printing these informations in console? This is an important information for the users.

**Comment by HolQue** [comment]
Another aspect: `variant` and `version_sw` _should_ be provided in command line. Otherwise the database application generates default values. Is this mechanism not active in append mode?

**Comment by ngoan1608** [comment]
> I do not understand this. If `variant`and `version_sw` is provided to the database application also in append mode, why not printing these informations in console? This is an important information for the users.

Hi Holger,

The execution result record in database is created with _UUID, variant, project, version_sw, version_hw_, ...
So that when using append mode, the above record is already existing with all above information and only the UUID is used for verification and no new execution record is created.

In case you use append mode and provide the different _variant, project, version_sw, version_hw_ information from the existing ones in database. The console log which displayed the given variant and version from command line args (or configuration file, or test metadata) is not make-sense because they are not used.

Thank you,
Ngoan

**Comment by HolQue** [comment]
OK, I understand. But the impact is that it's possible that a user provides command line arguments that are ignored. That's nit OK!

Possible solution: In case of append mode no other command line arguments must be allowed! Otherwise error message.

**Comment by test-fullautomation** [comment]
Hi Holger,
unfortunately I disagree to your last statement.
If the command line arguments are correct, means UUID, and versions are O.K., then the tool must run.
What we need is
1. log out all information provided by command line (e.g. also version information).
2. Check if version info is correct. If it deviates from previous upload with same UUID -> speaking error.
Thank you,
 Thomas

**Comment by HolQue** [comment]
Hey, cool:

```
Set project/variant to 'QT-PV-064' (from --variant commandline argument)
Set version_sw to 'QT-VSW-064' (from --versions commandline argument)
```

This helps :-)

**Comment by test-fullautomation** [comment]
solved and already merged with #26
